test/unittest/dds/conditions/ConditionsTests.cpp

```diff
@@ -31,6 +32,16 @@ TEST(ConsitionsTests, GuardConditionActivation)
     ASSERT_FALSE(cond.get_trigger_value());
 }
 
+TEST(ConsitionsTests, PSMGuardConditionActivation)
+{
+    ::dds::core::cond::GuardCondition cond;
+    ASSERT_FALSE(cond.trigger_value());
+    ASSERT_NO_THROW(cond.trigger_value(true));
```

Comment:
```suggestion
    cond.trigger_value(true);
```
I think it is not necessary as the PIM function only return RETCODE_OK, so there isn't any case in which it will throw an exception. The same for line 41.